tldw_Server_API/app/services/admin_budgets_service.py

```
@@ -463,10 +463,41 @@ def merge_budget_settings(
+            merged_providers = _merge_provider_budgets(merged.get(key), value)
+            if merged_providers is None:
+                merged.pop(key, None)
+            else:
+                merged[key] = merged_providers
+            continue
         raise ValueError("invalid_budget_update")
     return merged
 
 
+def _merge_provider_budgets(
+    existing: Any, updates: Any
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `existing`

(ANN401)